src/lib/y2storage/proposal/devices_planner.rb

```diff
@@ -93,21 +93,40 @@ def planned_boot_devices(planned_devices)
       # @param required_size [DiskSize]
       # @return [Partition]
       def reusable_swap(required_size)
-        return nil if settings.use_lvm || settings.use_encryption
+        return nil if skip_swap_reuse?
 
         partitions = available_swap_partitions
-        partitions.select! { |part| part.size >= required_size }
+        partitions.select! { |part| can_be_reused?(part, required_size) }
         # Use #name in case of #size tie to provide stable sorting
         partitions.min_by { |part| [part.size, part.name] }
       end
 
-      # Returns all avaiable swap partitions
+      # Returns all available and acceptable swap partitions
       #
       # @return [Array<Partition>]
       def available_swap_partitions
         devicegraph.partitions.select(&:swap?)
       end
 
+      # Whether reusing swap partitions is pointless
+      #
+      # @return [Boolean]
+      def skip_swap_reuse?
+        settings.use_lvm || settings.use_encryption || settings.swap_reuse == :none
+      end
+
+      # Whether it is acceptable to reuse the given swap partition
+      #
+      # @param partition [Partition]
+      # @param required_size [DiskSize]
+      # @return [Boolean]
+      def can_be_reused?(partition, required_size)
+        return false if partition.size < required_size
+        return true unless settings.swap_reuse == :candidate
```

Comment:
here I think it is confusing to match against candidate, where in fact you want to check if it is any so I would use `return true if settings.swap_reuse == :any`. Third `:none` cannot happen and here it is more easy to read like reuse it if anything is good enough.